### PR TITLE
Fix the navigation once and for all

### LINKED ISSUE #628: GitHub pane navigation broken
Now that the maintainer workflow branch has been merged to master, it's clear that navigation in the GitHub pane is broken: switching branches switches the pane back to the PR list and subsequent navigation with the back/forward/PR buttons is broken.

### LINKED ISSUE #631: New Commits & Files Changed Not Updated
# Description

After a commit is made and local branch is synced with remote, Contributor should be able to see the number of commits made and file changed when pane is refreshed but currently, the **Changes** section of the GitHub pane shows the files changed which do not correspond with the number of files changed and commits below the branch.
- GitHub Extension version: v2.0.15.3-alpha1 
- Visual Studio version: Enterprise 2015 -  Update 3

**Steps to reproduce**
- Have a repo that has an Open PR.
- Make sure branch is up-to-date
- Make changes to the branch and commit changes
- Click Team Explorer ==> Home Icon
- Click Sync and Push Outgoing commits
- Click Home Icon ==> Pull Requests
- Select the Open PR to see PR details
- Click Refresh icon to reflect the changes made.

**Expected** - New Commits and the number of files changed should correspond with the number of commits and files changed on dot com.

**Actual** - New Commits and the number of files changed are not reflected on the change when pane is refreshed. New commits and number of files changes are reflected after numerous Sync ==> Pull and Fetch of the branch which results to [#630](https://github.com/github/VisualStudio/issues/630)

### PR DESCRIPTION
**Pull request by shana**
Fixes the navigation in the GitHub pane.

Fixes #628 
Fixes #631

### CODE REVIEW

**Comment by rubenscholfield** [comment]
Yyyyyyeeeeeeeaaaaaaa, love you.
Team

Am 20.12.2016 9:04 nachm. schrieb "Andreia Gaita" <notifications@github.com

**Comment by rubenscholfield** [comment]
Thank you!!!

Am 24.12.2016 1:38 vorm. schrieb "Ruben Scholfield" <
futuretype2050@gmail.com>:

**Comment by rubenscholfield** [comment]
bitcoin:3P24jYtnsvW2bv9XJUeowCsWNxNKnYfvtZ?amount=5.214036

File: src/GitHub.App/ViewModels/PullRequestDetailViewModel.cs [hunk 1]
```diff
@@ -267,7 +267,7 @@ public string OperationError
         /// <param name="data"></param>
         public override void Initialize([AllowNull] ViewWithData data)
         {
-            var prNumber = (int)data.Data;
+            var prNumber = data?.Data != null ? (int)data.Data : Model.Number;
```
Comment (shana) [inline src/GitHub.App/ViewModels/PullRequestDetailViewModel.cs]: Aaah, yes, nice catch, that's what was missing!

File: src/GitHub.App/Controllers/NavigationController.cs [hunk 2]
```diff
@@ -14,7 +15,7 @@ namespace GitHub.Controllers
     public class NavigationController : NotificationAwareObject, IDisposable, IHasBusy
     {
         readonly List<IUIController> history = new List<IUIController>();
-        readonly Dictionary<UIControllerFlow, IUIController> controllers = new Dictionary<UIControllerFlow, IUIController>();
+        readonly Dictionary<UIControllerFlow, IUIController> reusableControllers = new Dictionary<UIControllerFlow, IUIController>();
```
Comment (shana) [inline src/GitHub.App/Controllers/NavigationController.cs]: :heart:

**Comment by shana** [comment]
This looks great! 🚀

**Pull request merged**